containers/src/Kokkos_OffsetView.hpp

```diff
@@ -180,30 +180,28 @@ void runtime_check_rank_device(const size_t rank_dynamic, const size_t rank,
 }  // namespace Impl
 
 template <class DataType, class... Properties>
-class OffsetView : public ViewTraits<DataType, Properties...> {
- public:
-  using traits = ViewTraits<DataType, Properties...>;
-
+class OffsetView : public View<DataType, Properties...> {
  private:
   template <class, class...>
   friend class OffsetView;
-  template <class, class...>
-  friend class View;  // FIXME delete this line
-  template <class, class...>
-  friend class Kokkos::Impl::ViewMapping;
 
-  using map_type   = Kokkos::Impl::ViewMapping<traits, void>;
-  using track_type = Kokkos::Impl::SharedAllocationTracker;
+  using base_t = View<DataType, Properties...>;
 
  public:
-  enum { Rank = map_type::Rank };
-  using begins_type = Kokkos::Array<int64_t, Rank>;
+  // typedefs to reduce typing base_t:: further down
+  using traits = typename base_t::traits;
+  // FIXME: should be base_t::index_type after refactor
+  using index_type   = typename base_t::memory_space::size_type;
```

Comment:
Maybe move `index_type` to private, as it was not exposed in public in the previous implementation?
Also, previously, `reference_type` was part of the public interface.

Reply:
`OffsetView` is derived from `Kokkos::View` so it gets `reference_type` from there. I will actually submit a PR which introduces `index_type` for `Kokkos::View` (the legacy version). The new version will anyway have it because I am bringing it in line with `mdspan`.